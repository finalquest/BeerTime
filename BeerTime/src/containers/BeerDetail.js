import { connect } from 'react-redux';
import { back } from '../model/actions/navigation';
import BeerDetailScreen from '../screens/BeerDetailScreen';

const mapStateToProps = (_, { navigation: { getParam } }) => ({
  beer: getParam('beer'),
});
const mapDispatchToProps = dispatch => ({
  onBack: () => dispatch(back()),
});
export default connect(mapStateToProps, mapDispatchToProps)(BeerDetailScreen);
