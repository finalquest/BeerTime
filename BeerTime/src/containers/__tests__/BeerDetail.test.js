import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import { Linking } from 'react-native';
import BeerDetail from '../BeerDetail';


const mockStore = configureStore();
const initialState = {};

const navigation = {
  getParam: () => undefined,
};

jest.mock('Linking', () => {
  const onCall = jest.fn();
  return {
    openURL: onCall,
    canOpenURL: () => true,
  };
});

describe('BeerList test suite', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <BeerDetail navigation={navigation} />,
      { context: { store: mockStore(initialState) } },
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should dispatch back', () => {
    const store = mockStore(initialState);
    const wrapper = shallow(
      <BeerDetail navigation={navigation} />,
      { context: { store } },
    );
    wrapper.find('BeerDetailScreen').prop('onBack')();
    expect(store.getActions()).toEqual([{ type: 'Navigation/BACK', key: null, immediate: undefined }]);
  });

  test('should open browser', () => {
    const store = mockStore(initialState);
    const wrapper = shallow(
      <BeerDetail navigation={navigation} />,
      { context: { store } },
    );
    wrapper.find('BeerDetailScreen').prop('openBrowser')('test');
    expect(Linking.openURL).toBeCalledWith('https://www.allrecipes.com/search/results/?wt=test');
  });
});
