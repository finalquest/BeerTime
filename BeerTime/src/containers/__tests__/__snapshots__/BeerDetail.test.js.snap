// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerList test suite renders as expected 1`] = `
<BeerDetailScreen
  navigation={
    Object {
      "getParam": [Function],
    }
  }
  onBack={[Function]}
/>
`;
